crates/trie/sparse-parallel/src/metrics.rs

```diff
@@ -0,0 +1,23 @@
+//! Metrics for the parallel sparse trie
+use reth_metrics::{metrics::Histogram, Metrics};
+
+/// Metrics for the parallel sparse trie
+#[derive(Metrics, Clone)]
+#[metrics(scope = "parallel_sparse_trie")]
+pub(crate) struct ParallelSparseTrieMetrics {
+    /// A histogram for the number of subtries updated when calculating hashes.
+    pub(crate) subtries_updated: Histogram,
+    /// A histogram for the time it took to update lower subtrie hashes.
+    pub(crate) subtrie_hash_update_latency: Histogram,
+    /// A histogram for the time it took to update the upper subtrie hashes.
+    pub(crate) subtrie_upper_hash_latency: Histogram,
+}
+
+impl PartialEq for ParallelSparseTrieMetrics {
```

Comment:
ah this is because we derive this for ParallelSparseTrie, do we even this this derive there?
since this is pub crate this is fine tho

Reply:
> do we even this this derive there?

its mainly for tests
https://github.com/paradigmxyz/reth/blob/3772535220784420370fff9c13f85d17d15eb786/crates/trie/sparse-parallel/src/trie.rs#L4623